### PR TITLE
DISCOVERYACCESS-8168: Fix bug preventing some Aeon links from working

### PR DESCRIPTION
**Pull request by Baroquem**
https://culibrary.atlassian.net/browse/DISCOVERYACCESS-8168

This is a weird one. It turns out that the Aeon links that aren't working are having hidden form values added to them via [this Coffeescript](https://github.com/cul-it/blacklight-cornell/blob/1911503652fd2bec4e897a1c910d1a99fe6780be/app/assets/javascripts/holdings.js.coffee#L188-L213). This is apparently done to pass some extra parameters to Rails along with the original GET request -- e.g., the `counter` variable, which is used throughout the code in various forms. Rather than try to disentangle all of that, I took the expedient approach of adding `PUT` to the methods allowed for this particular route. This should be a safe, albeit inelegant, solution, now that I understand what's going on.

The only important change is in routes.rb. I made a few minor edits in other files to clean up the code a bit; then, VS Code got into the spirit of the thing and decided to fix a lot of formatting on its own. That's why cornell_catalog_helper has so many tab adjustments!

### CODE REVIEW

**Review by mhk33 — APPROVED** [review]

File: app/helpers/cornell_catalog_helper.rb [hunk 1]
```diff
@@ -1432,31 +1432,27 @@ def sub_tilde_path(var,id,aeon_codes,document)
   end
 
 # (group == "Circulating" ) ? blacklight_cornell_request.magic_request_path("#{id}") :  "http://wwwdev.library.cornell.edu/aeon/monograph.php?bibid=#{id}&libid=#{aeon_codes.join('|')}"
-  def request_path(group,id,aeon_codes,document,scan)
-	id_scan = scan == "yes" ? "#{id}.scan" : "#{id}"
+  def request_path(group, id, aeon_codes, document, scan)
+    id_scan = scan ? "#{id}.scan" : "#{id}"
```
Comment (chrisrlc) [inline app/helpers/cornell_catalog_helper.rb:1436]: I'm guessing this means that `scan` won't ever have a value that should be treated as no/false?

Comment (chrisrlc) [reply app/helpers/cornell_catalog_helper.rb:1436]: Oops, I see your code below now. Ignore this. Great refactor.

File: app/helpers/cornell_catalog_helper.rb [hunk 2]
```diff
@@ -1432,31 +1432,27 @@ def sub_tilde_path(var,id,aeon_codes,document)
...
     magic_path  = blacklight_cornell_request.magic_request_path("#{id_scan}")
+
     if ENV['SAML_IDP_TARGET_URL'].present?
-    magic_path = blacklight_cornell_request.auth_magic_request_path("#{id_scan}")
+      magic_path = blacklight_cornell_request.auth_magic_request_path("#{id_scan}")
     end
+
     if ENV['AEON_REQUEST'].blank? and group != 'AEON_SCAN_REQUEST'
```
Comment (chrisrlc) [inline app/helpers/cornell_catalog_helper.rb:1443]: Optional, but maybe we could also change `and` to `&&`? Reduces possibility of error in any future refactors due to confusingly different levels of precedence in ruby.

**Review by chrisrlc — APPROVED** [review]

File: app/helpers/cornell_catalog_helper.rb (hunk 2 above)
Comment (Baroquem) [reply app/helpers/cornell_catalog_helper.rb]: Good idea!

**Pull request merged**